### PR TITLE
Ticket transfers

### PR DESCRIPTION
**Pull request by melinath**
This is an incomplete implementation of ticket transfers (#332) which I am making into a pull request because I have forgotten what still needs to be done. I almost forgot the branch existed. o_O As I figure it out I will add notes here.

So. Things that need to work:

* [x] User can initiate a transfer of an item
* [x] Target can accept transfer (via the invite system)
* [x] Item is still recorded at price for former user
* [x] Item is also recorded for new user, but at $0
* [x] If an attendee has no more items, delete them
* [x] Admins can turn ticket transfers on and off

Things to create separate tickets for:

* [ ] User can cancel a transfer of an item
* [ ] Add tests of get_summary_data
* [ ] Freeze attendee info after checkout (i.e. no implicit transfers.) Link to transfer from attendee detail page.

### CODE REVIEW

**Comment by melinath** [comment]
So I've looked over this, and I found one issue, but I think this is otherwise ready for merge, actually.

File: brambling/models.py [hunk 1]
```diff
@@ -739,6 +727,126 @@ def get_icon(self):
...
+                    event=event,
+                    person=request.user,
+                )
+            except Order.DoesNotExist:
+                pass
+
+        # Next, check if there's a session-stored order. Assign it
+        # if the order hasn't checked out yet and the user is authenticated.
+        if order is None:
+            code = self._get_session_code(self.request, event)
```
Comment (chigby) [inline brambling/models.py]: I think `self.request` should just be `request`.

**Comment by chigby** [comment]
Other than that one code comment, I'd say this is ready to merge.

File: brambling/models.py (hunk 1 above)
Comment (melinath) [inline brambling/models.py]: Yes. Yes it should. :-/

Comment (melinath) [inline brambling/models.py]: Fixed plus tests!

**Comment by melinath** [comment]
Ready for re-review!

**Pull request merged**